hoomd/pytest/test_simulation.py

```diff
@@ -0,0 +1,172 @@
+import hoomd
+import numpy as np
+import pytest
+from copy import deepcopy
+try:
+    import gsd.hoomd
+    skip_gsd = False
+except ImportError:
+    skip_gsd = True
+
+
+@pytest.fixture(scope="function")
+def get_snapshot(device):
+    def make_snapshot(n=10, particle_types=['A']):
+        s = hoomd.snapshot.Snapshot(device.comm)
+        if s.exists:
+            s.configuration.box = [20, 20, 20, 0, 0, 0]
+            s.particles.N = n
+            s.particles.position[:] = np.random.uniform(-10, 10, size=(n, 3))
+            s.particles.types = particle_types
+        return s
+    return make_snapshot
+
+
+def make_gsd_snapshot(hoomd_snapshot):
+    s = gsd.hoomd.Snapshot()
+    snap_properties = []
+    all_attr = dir(hoomd_snapshot)
+    for att in all_attr:
+        if att[0] != '_' and att not in ['exists', 'replicate']:
+            snap_properties.append(att)
```

Comment:
This is fine, but it may be a good opportunity to use the list comprehension syntax. I don't care too much either way though.

Reply:
Disregard this comment.